### PR TITLE
Updated the pettingzoo.tests.performance_benchmark function to display output as mean ± stdev

### PR DESCRIPTION
**Pull request by ndyashas**
to make multiple smaller runs and report results in the
mean ± stdev format.

Commit related to #170.

### CODE REVIEW

File: pettingzoo/tests/performance_benchmark.py [hunk 1]
```diff
@@ -24,14 +25,59 @@ def performance_benchmark(env):
...
+        # Warn if the maximum is 50% greater than the mean
+        print(f"[WARNING] for {name} maximum is 50% greater than the mean", file=sys.stderr)
+
+
+def performance_benchmark(env):
+    print("Starting performance benchmark")
+
+    test_time = 5
+    test_sample_time = 1
+    check_benchmark_stability = True
```
Comment (ndyashas) [inline pettingzoo/tests/performance_benchmark.py:64]: I have left these parameters internally and not added them as arguments list so as to not break the current API. However, if added to the argument list, they still will be backward compatible.

Comment (weepingwillowben) [reply pettingzoo/tests/performance_benchmark.py:64]: It should be fine to add these to the arguments list

Comment (justinkterry) [reply pettingzoo/tests/performance_benchmark.py:64]: make sure you update the documentation accordingly

**Comment by justinkterry** [comment]
I don't understand why this is functionality that we need or would want to add

**Comment by ndyashas** [comment]
With the existing implementation, multiple runs report different benchmarking values without an estimation of how much error is present in them. The primary idea was to report the result along with an estimated error which would give a better idea than just the absolute value.

This modification does not significantly improve functionality. I made these with the following in mind.

- The existing implementation uses ```time.time()``` which is similar to wall-clock time and varies depending on other processes running on the system at the time, instead ```time.process_time()``` ignores the time for which the current process was put to sleep by the OS, thus a better measure compared to ```time.time()```.
- Splitting the runs allows us to estimate if the benchmarking was more-or-less consistent as it is done in the ```pyperf``` tool [here](https://pyperf.readthedocs.io/en/latest/cli.html#pyperf-check).
- Splitting the runs also lets us report the result along with estimated error. eg: ```mean ± stdev```.

**Comment by weepingwillowben** [comment]
So I looked at the standard deviations reported by travis's run, and it was incredibly small. Less than 10% in all cases, if your code was right. So I guess I was right about it being small. 

We feel that users are more concerned about factors of 2 or more, as the systems which use these environments never run in real time and are never very sensitive to execution time. 

Also, we feel that time.time() is the better and more useful measure because it can handle all weirdness that may go on behind the scenes of an environment such as:

1. Environments spawning off sub-processes which do the real work (none of the current pettingzoo environments do this, but the starcraft micromanagement environments do).
1. Multiple threads (magent used to do that, I disabled it)
1. Any other weirdness that will put the process clock out of alignment with what people actually care about: wall time. 


And when you are talking about multiple seconds, the amount of sleeping that happens should be relatively small unless the system is completely overloaded.  So process time solves one problem but introduces potentially larger problems.

**Comment by ndyashas** [comment]
Ah, yes. The points make sense. Thank you very much for your time and the detailed explanation!. I will close this PR and the related Issue for now, as it makes more sense to continue with the current functionality as you mentioned.

**Pull request closed without merging**